src/BasicAbility/Masa.Contrib.BasicAbility.Pm/PmCaching.cs

```diff
@@ -0,0 +1,73 @@
+
+namespace Masa.Contrib.BasicAbility.Pm
+{
+    public class PmCaching : IPmCaching
+    {
+        private readonly IMemoryCacheClient _memoryCacheClient;
+        private readonly ICallerProvider _callerProvider;
+
+        public PmCaching(IMemoryCacheClient memoryCacheClient, ICallerProvider callerProvider)
+        {
+            _memoryCacheClient = memoryCacheClient;
+            _callerProvider = callerProvider;
+        }
+
+        public async Task<List<Project>> GetProjectListAsync(string envName)
+        {
+            var keys = await _memoryCacheClient.GetAsync<List<string>>($"{PROJECT_KEY_PREFIX}.{envName.ToLower()}");
+            if (keys == null)
+            {
+                return new List<Project>();
+            }
+            else
+            {
+                var projects = await _memoryCacheClient.GetListAsync<Project>(keys.ToArray());
+                var result = projects.Where(project => project != null).ToList();
+
+                return result!;
+            }
+        }
+
+        public async Task<List<App>> GetAppListAsync(string envName)
```

Comment:
GetAppListAsync means GetAppListByProjectAsync, maybe can call `GetAllAppsAsync`, and why need this api?